dynawo/sources/Modeler/DataInterface/PowSyblIIDM/test/TestLoadInterface.cpp

```diff
@@ -93,32 +93,63 @@ TEST(DataInterfaceTest, Injector) {
   ASSERT_TRUE(loadIfce.getInitialConnected());
   load.getTerminal().disconnect();
   ASSERT_TRUE(loadIfce.getInitialConnected());
-  {
+
+  ASSERT_EQ(loadIfce.getBusInterface().get(), nullptr);
+  const boost::shared_ptr<BusInterface> busIfce(new BusInterfaceIIDM(bus1));
+  loadIfce.setBusInterface(busIfce);
+  ASSERT_EQ(loadIfce.getBusInterface().get()->getID(), "VL1_BUS1");
+
+  ASSERT_FALSE(loadIfce.hasP());  // GGGF --> normal
+  ASSERT_FALSE(loadIfce.hasQ());  // GGGF
+
+  {  // tests assuming getInitialConnected == false
     LoadInterfaceIIDM connectedLoadIfce(load);
     ASSERT_FALSE(connectedLoadIfce.getInitialConnected());
     load.getTerminal().connect();
     ASSERT_FALSE(connectedLoadIfce.getInitialConnected());
+
+    ASSERT_FALSE(connectedLoadIfce.hasP());
+    ASSERT_FALSE(connectedLoadIfce.hasQ());
     ASSERT_DOUBLE_EQ(connectedLoadIfce.getP(), 0.0);
     ASSERT_DOUBLE_EQ(connectedLoadIfce.getQ(), 0.0);
+
+    load.getTerminal().setP(100000.0);
+    ASSERT_TRUE(connectedLoadIfce.hasP());
+    ASSERT_DOUBLE_EQ(connectedLoadIfce.getP(), 0.0);
+
+    load.getTerminal().setQ(1000000000.0);
+    ASSERT_TRUE(connectedLoadIfce.hasQ());
+    ASSERT_DOUBLE_EQ(connectedLoadIfce.getQ(), 0.0);
+
+    load.getTerminal().disconnect();
+    ASSERT_TRUE(connectedLoadIfce.hasP());  // GGGF --> weird since disconnected
+    ASSERT_TRUE(connectedLoadIfce.hasQ());
   }
 
-  ASSERT_EQ(loadIfce.getBusInterface().get(), nullptr);
-  const boost::shared_ptr<BusInterface> busIfce(new BusInterfaceIIDM(bus1));
-  loadIfce.setBusInterface(busIfce);
-  ASSERT_EQ(loadIfce.getBusInterface().get()->getID(), "VL1_BUS1");
+  ASSERT_TRUE(loadIfce.hasP());  // It was 'false' before connect/disconnect operations! // GGGF
```

Comment:
I don't understand why this brutally becomes true

Reply:
That is what I was trying to explain. loadIfce and connectedLoadIfce are 2 interfaces on the same terminal. connectedLoadInterface was a local interface for a special test on P and Q. I set P on the terminal, not on the interface. When the interface deasappears, the terminal keeps having some P and Q. Then what was FALSE through the interface before connecting/disconnecting the terminal locally comes to be true now. That what I noticed to be weird. I will pass the test only wy checking 2 interfaces on different terminals. I don't know enough of electrical injectors to understand the logical stuff, but it comes :)

Reply:
I see you cannot compile because it lacks the .h file. Sorry. It compiles but test does not work since voltage found on P remains... I had the file

Reply:
Then the comment is false: it becomes true not because of the (dis)connection but because a P and Q were defined. Please adapt the comment

Reply:
done